packages/calcite-components/src/utils/date.ts

```diff
@@ -151,7 +151,10 @@ export function datePartsFromLocalizedString(
  */
 export function dateToISO(date?: Date): string {
   if (date instanceof Date) {
-    return new Date(date.getTime() - date.getTimezoneOffset() * 60000).toISOString().split("T")[0];
+    const month = String(date.getMonth() + 1).padStart(2, "0");
```

Comment:
Could this be simplified to

```ts
return date instanceof Date ? date.toISOString().split('T')[0] : "";
```

?

Reply:
the `toISOString` is the problematic function, so that wouldn't fix the issue.

Reply:
![image](https://github.com/Esri/calcite-design-system/assets/821864/0363dac9-8f1c-4b18-933e-299f3efd09b7)

Reply:
1. Yes, the docs should be more clear on what it returns exactly.  I actually think the name `dateToISO` is appropriate in this case because when you look up the [ISO 8601 standard](https://en.wikipedia.org/wiki/ISO_8601), it does call out a valid way to express just the date portion here: 
![image](https://github.com/Esri/calcite-design-system/assets/821864/9a6ba0d8-a61a-4620-b7cb-bf88ad74dbc4)
2. Will do.  I'm also wondering if a comment in the code of the function could be worth it as well to remember the decision why we moved away from `toISOString`?
3. Yes.
4. Could be worth it, but because the `Date` object's instance methods always return values in the local timezone (`getDate`, `getMonth`, `getFullYear`) it might be overkill to introduce a plugin for something its already handling.

Generally, the issue I've found that I can't seem to workaround is exactly how to get the exact `date` value back out of the constructed `Date` object that was passed in.  Because all the `get*` methods return in local time, if the `Date` was constructed with a CEST timezone offset specified in the input string like:

```
const cestDate = new Date("1850-03-13T00:00:00.000+02:00");
```
There's no way to retrieve the value `13` from that original string back out of the `Date` object in a reliable way that is independent of the current system timezone setting.